src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs

```diff
@@ -128,14 +141,25 @@ await foreach (var set in allSets)
                                 }
                             }
 
-                            if (priority == CodeActionRequestPriority.Normal)
+                            // We're finishing up with a particular priority group, and we're about to go to a priority
+                            // group one lower than what we have (hence `priority - 1`).  Take any pending items in the
+                            // group we're *about* to go into and add them at teh end of this group.
+                            //
+                            // For example, if we're in the high group, and we have an pending items in the normal
+                            // bucket, then add them at the end of hte high group.  The reason for this is that we
+                            // already have computed the items and we don't want to force them to have to wait for all
+                            // the processing in their own group to show up.  i.e. imagine if we added at teh start of
+                            // the next group.  They'd be in the same location in the lightbulbt as when we add at the
+                            // end of the current group, but they'd show up only when that group totally finished,
+                            // instead of right now.
+                            //
+                            // This is critical given that the lower pri groups are often much lower (which is why they
+                            // they choose to be in that class).  We don't want a fast item computed by a higher pri
+                            // provider to still have to wait on those slow items.
```

Comment:
Nit:
```suggestion
                            // We're finishing up with a particular priority group, and we're about to go to a priority
                            // group one lower than what we have (hence `priority - 1`).  Take any pending items in the
                            // group we're *about* to go into and add them at the end of this group.
                            //
                            // For example, if we're in the high group, and we have an pending items in the normal
                            // bucket, then add them at the end of the high group.  The reason for this is that we
                            // already have computed the items and we don't want to force them to have to wait for all
                            // the processing in their own group to show up.  i.e. imagine if we added at the start of
                            // the next group.  They'd be in the same location in the lightbulb as when we add at the
                            // end of the current group, but they'd show up only when that group totally finished,
                            // instead of right now.
                            //
                            // This is critical given that the lower pri groups are often much lower (which is why they
                            // they choose to be in that class).  We don't want a fast item computed by a higher pri
                            // provider to still have to wait on those slow items.
```